File: libs/document-processor/src/repo/core/schema/SequentialWorkflow.ts
import { z } from 'zod';
import { blueIds } from '../blue-ids';
import { withTypeBlueId, blueNodeField } from '@blue-labs/language';

export const SequentialWorkflowSchema = withTypeBlueId(
  blueIds['Sequential Workflow']
)(
  z.object({
    steps: z.array(blueNodeField()).optional(),
    channel: z.string().optional(),
  })
);

export type SequentialWorkflow = z.infer<typeof SequentialWorkflowSchema>;

File: references/java-sdk/src/main/java/blue/language/types/conversation/SequentialWorkflow.java
package blue.language.types.conversation;

import blue.language.model.Node;
import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;

import java.util.List;

@TypeAlias("Conversation/Sequential Workflow")
@TypeBlueId("7X3LkN54Yp88JgZbppPhP6hM3Jqiqv8Z2i4kS7phXtQe")
public class SequentialWorkflow {
    public List<Node> steps;

    public SequentialWorkflow steps(List<Node> steps) {
        this.steps = steps;
        return this;
    }
}
